### ISSUE TITLE
nvme I/O timeout, aborting warning messages in journal on Fedora guest VM Windows host

### ISSUE THREAD

**Issue by hermidalc** [vmware/open-vm-tools#579]
### Describe the bug

I see quite often warnings messages like `nvme nvme0: I/O 34 QID 13 timeout, aborting` in the journal and sometimes it correlates with I/O appearing to hang a bit. Why is this happening?

My Fedora guest VM is running on an NVMe disk on latest VMware Workstation 16.2.3 Windows 10 host. I am using the latest firmware and drivers for the NVMe disk.

### Reproduction steps

```bash
Hard to show how to reproduce
```

### Expected behavior

No nvme timeout warning messages

**Comment by PaTHml** [comment]
NVMe storage issues are not related to open-vm-tools;
Please open a service/support request with VMware support to diagnose the issue further.

**Comment by hermidalc** [comment]
> NVMe storage issues are not related to open-vm-tools;
> Please open a service/support request with VMware support to diagnose the issue further.

VMware support is generally a big waste of time for us customers, because they will think it’s a problem with your specific computer or setup and not a general issue with the software. 

They also don’t give different expedited support to power users who know it’s not a problem with their computer or setup. I didn’t have this issue historically on the same computer, same NVMe drive, and same Windows install. I’ve seen this issue posted by many other users online. That’s why I don’t bother with VMware Support because I don’t want to spend cycles proving to them that it’s an issue on their side not mine.

**Comment by PaTHml** [comment]
As an aside, the VMTN for Workstation has/had a similar issue thread, might be of help to you:
https://communities.vmware.com/t5/VMware-Workstation-Pro/Workstation-Pro-16-NVMe-controller/m-p/2822786#M168257

**Comment by hermidalc** [comment]
> As an aside, the VMTN for Workstation has/had a similar issue thread, might be of help to you: https://communities.vmware.com/t5/VMware-Workstation-Pro/Workstation-Pro-16-NVMe-controller/m-p/2822786#M168257

Exactly, I already saw this and multiple other threads online of users having the same problem. That's why I don't want to deal with VMware support because they will make me go through wasted time and cycles assuming first it's specific to my box and setup when clearly it's not it's a general VMware issue.

Summary from the thread above - something is generally going wrong with the VMware virtual NVMe adapter and it needs to be fixed.

**Comment by RevAngel7** [comment]
Hello. 
I am not using VMware, but I have this issue on ubuntu with the latest mainline kernels.

I am using 5.19 and have that issue from 5.19.1 to the now recent, and used by me, 5.19.11 from ubuntu 22.04 mainline.

nvme nvme0: Abort status: 0x0
nvme nvme0: I/O 14 QID 2 timeout, aborting
nvme nvme0: Abort status: 0x0
nvme nvme0: I/O 62 QID 2 timeout, aborting
and so on...

I have these issues after I changed from a AMD 2400G with an AM3 AGESA (on X370 chipset) before 1.0.0.6 to a AMD 5600G on a AM3 with AGESA 1.2.0.7 (on A520 chipset). NVME drive is the same. 

I also get an error message at bootup from the NVME:
Device: /dev/nvme0, number of Error Log entries increased from 203 to 206
This counter rises +1 every poweroff (203-206 comes from an image backup I did with 3 manual power off's, so it seems every power off counts as one error count).

The NVME drive never breached the high temp count and shows zero errors and very little wear on SMART tests, since I use this drive for a simple "daily use" multimedia system.

When the "nvme nvme0: Abort status: 0x0" / "nvme nvme0: I/O 14 QID 2 timeout, aborting" errors occur, the system hangs for a while, no I/O operations get processed for up to 30 seconds.

Since this entry is from 2020 and still an issue in late 2022, is there a global fix or a planned fix for the user or planned to kernel changes?

Thank you for any reply and/or ideas.

**Comment by johnwvmw** [comment]
@RevAngel7  From your statement

`I am not using VMware, but I have this issue on ubuntu with the latest mainline kernels.`

it suggests that the problem(s) are not Workstation or open-vm-tools issues but something tied directly to the Linux kernel release and/or the NVME driver.  Thanks for the 2020 bug reference.

Has the problem been raised with the Linux vendor(s)?

**Comment by RevAngel7** [comment]
I totally get why my report is out of place. I really do. And since I consider myself more of a user than a tech savy person I also understand the reluctance to consider my comment a real issue.

This bug is still open, if I am reading it right.

The same issue on https://github.com/vmware/open-vm-tools/issues/579 , also unsolved.

The same issue on https://github.com/clearlinux/distribution/issues/2121 , also unsolved.

And there is my issue on ubuntu.

Three different kernels, linux brands, same issue. I thought bringing the people together who actually have the tech knowledge to get behind this issue might be helpful (but that's just me).

**Comment by hermidalc** [comment]
FYI this bug still occurs on Fedora 36 guest with Linux kernal 5.19, so not sure if it's a kernel bug

**Comment by RevAngel7** [comment]
I don't even know where to report this for ubuntu, to be totally honest. Like I said, I am a user who just stumbled over his ubuntu logs and googled them, found the entries here and Clear Linux distro. Do you have any suggestion what to do to help solving this riddle?

**Comment by RevAngel7** [comment]
Posted it on https://bugs.launchpad.net/launchpad/+bug/1991291 fyi

**Comment by johnwvmw** [comment]
Launchpad is the bug tracking system used by Ubuntu.   Also a search of "Ubuntu how to file a problem report" led me to https://help.ubuntu.com/community/ReportingBugs.   This provides some additional info about bug reporting tools that make it easy to capture crash dumps and system dumps for upload, if needed.

**Comment by hermidalc** [comment]
Actually I'm probably wrong, I've search for this issue across Google and see it mentioned in multiple Linux dists without mentioning VMware. So yes could be a Linux kernel issue. Others have also mentioned still present in 5.19

**Comment by SIMULATAN** [comment]
Sorry for necroposting, I personally experience this issue on stock Arch on physical hardware, also since ~5.19, so +1 for a linux kernel bug from me.
(it just happened to me on 6.0.8, the issue seems to still be present nowadays)

**Comment by OSS542** [comment]
This is still an issue in kernel 6.1 - see https://bugzilla.kernel.org/show_bug.cgi?id=216809

**Comment by adozenlines** [comment]
Also under Fusion.
